connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/entities/ConnectorStateInfo.java

```diff
@@ -103,6 +104,21 @@ public int id() {
         public int compareTo(TaskState that) {
             return Integer.compare(this.id, that.id);
         }
+
+        @Override
+        public boolean equals(Object o) {
+            if (o == this)
+                return true;
+            if (!(o instanceof AbstractState))
```

Comment:
You'll hate me, but I see a tiny chance for `ClassCastException` that we can avoid.